sql/core/src/test/scala/org/apache/spark/sql/parquet/ParquetSchemaSuite.scala

```diff
@@ -109,20 +245,21 @@ class ParquetSchemaSuite extends SparkFunSuite with ParquetTest {
       |    optional binary _2 (UTF8);
       |  }
       |}
-    """.stripMargin)
+    """.stripMargin,
+    followParquetFormatSpec = true)
 
-  testSchema[Tuple1[Map[Int, (String, Seq[(Int, Double)])]]](
-    "deeply nested type",
+  testSchemaInference[Tuple1[Map[Int, (String, Seq[(Int, Double)])]]](
+    "deeply nested type - non-standard",
     """
       |message root {
-      |  optional group _1 (MAP) {
-      |    repeated group map (MAP_KEY_VALUE) {
+      |  optional group _1 (MAP_KEY_VALUE) {
```

Comment:
This test case is for non-standard behavior of old Spark versions, which mimic parquet-hive. There's a `followParquetFormatSpec` argument in `testSchemaInference`, which defaults to `false`. I should probably remove the default argument value and state explicitly here.